File: src/main/java/toby/lavaplayer/GuildMusicManager.java
package toby.lavaplayer;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager;

public class GuildMusicManager {
    private final AudioPlayer audioPlayer;
    private final TrackScheduler scheduler;
    private final AudioPlayerSendHandler sendHandler;


    public GuildMusicManager(AudioPlayerManager manager) {
        this.audioPlayer = manager.createPlayer();
        this.scheduler = new TrackScheduler(this.getAudioPlayer());
        this.getAudioPlayer().addListener(this.getScheduler());
        this.sendHandler = new AudioPlayerSendHandler(this.getAudioPlayer());
    }

    public AudioPlayerSendHandler getSendHandler(){
        return sendHandler;
    }

    public AudioPlayer getAudioPlayer() {
        return audioPlayer;
    }

    public TrackScheduler getScheduler() {
        return scheduler;
    }
}

File: src/main/kotlin/toby/lavaplayer/GuildMusicManager.kt
package toby.lavaplayer

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager

class GuildMusicManager(manager: AudioPlayerManager) {
    @JvmField
    val audioPlayer: AudioPlayer = manager.createPlayer()
    @JvmField
    val scheduler: TrackScheduler = TrackScheduler(this.audioPlayer)
    @JvmField
    val sendHandler: AudioPlayerSendHandler

    fun getAudioPlayer() = audioPlayer
    fun getScheduler() = scheduler
    fun getSendHandler() = sendHandler

    init {
        audioPlayer.addListener(this.scheduler)
        this.sendHandler = AudioPlayerSendHandler(this.audioPlayer)
    }
}
